scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,68 +116,167 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
         };
     }
 
-    List<Issue> runExternalAnalyzers(Project project) {
+    Optional<Map<String, List<Rule>>> getExternalAnalyzers(PrintStream outputStream) {
         StringBuilder newImports = new StringBuilder();
         StringBuilder tomlDependencies = new StringBuilder();
-
-        for (ScanTomlFile.Analyzer analyzer : scanTomlFile.getAnalyzers()) {
-            extractAnalyzerImportsAndDependencies(analyzer, newImports, tomlDependencies);
+        Set<ScanTomlFile.Analyzer> analyzers = scanTomlFile.getAnalyzers();
+        List<String> analyzerDescriptors = new ArrayList<>(analyzers.size());
+        for (ScanTomlFile.Analyzer analyzer : analyzers) {
+            analyzerDescriptors.add(extractAnalyzerImportsAndDependencies(analyzer, newImports, tomlDependencies));
         }
+
         Module defaultModule = project.currentPackage().getDefaultModule();
-        List<String> defaultModuleFiles = defaultModule.documentIds()
-                .stream()
-                .map(documentId -> defaultModule.document(documentId).name())
-                .toList();
-        String documentName;
-        do {
-            documentName = String.format("%s-%s.bal", IMPORT_GENERATOR_FILE, UUID.randomUUID());
-        } while (defaultModuleFiles.contains(documentName));
         ModuleId defaultModuleId = defaultModule.moduleId();
-        DocumentId documentId = DocumentId.create(documentName, defaultModuleId);
-        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(), documentName);
+        DocumentId documentId = DocumentId.create(generatedDocumentName, defaultModuleId);
+        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(), generatedDocumentName);
         defaultModule.modify().addDocument(documentConfig).apply();
-
         project.currentPackage().ballerinaToml().ifPresent(ballerinaToml -> {
             String tomlFileContent = ballerinaToml.tomlDocument().textDocument().toString();
             ballerinaToml.modify().withContent(tomlFileContent + tomlDependencies).apply();
         });
 
-        List<ScannerContext> scannerContextList = new ArrayList<>();
+        PackageResolution packageResolution = project.currentPackage().getResolution();
```

Comment:
This variable seems redundant.

Reply:
Shall we extract `project.currentPackage()` to a variable

Reply:
When engaging compiler plugins through the project API, the following steps are followed:

1. Generate an in-memory Ballerina file in the default module with the compiler plugins generated as imports.
2. Modify the in-memory `Ballerina.toml` file by generating the local compiler plugin toml dependencies.
3. Engage the generated compiler plugins through package compilation.

However, the project API allows doing one in-memory modification step at a time, and the subsequent steps to the modified project are applied only if the `currentPackage` is loaded again from the `project` object.

![image](https://github.com/ballerina-platform/static-code-analysis-tool/assets/87777949/4f68dfb7-e8e9-488b-9942-054e476d46bd)


Therefore we cannot extract the `currentPackage` to a re-usable variable.

Reply:
Got it!. It means you are changing the current-package separately and you expect that change.